include/engine/guidance/route_step.hpp

```diff
@@ -29,7 +29,26 @@ namespace guidance
 // Departure: s --> a --> b. Represents the segment s,a with location being s.
 // Arrive: a --> b --> t. The segment (b,t) is already covered by the previous segment.
 
-// A representation of intermediate intersections
+// A representation of intermediate intersections:
+// An intersection is an ordered list of connected roads ordered from from sharp right
+// counter-clockwise to sharp left where `intersection[0]` is always a u-turn
```

Comment:
the intersection in the output is a different object than the intersection in the preprocessing. Here the concept of `Connected Roads` is unknown. We offer available turn bearings (in all directions, not angles) and allowed entry flags.